library/ssl_tls.c

```diff
@@ -2895,13 +2883,24 @@ int mbedtls_ssl_flight_transmit( mbedtls_ssl_context *ssl )
         }
         else
         {
+            const int ret_payload = mbedtls_ssl_get_max_out_record_payload( ssl );
+            const size_t max_record_payload = (size_t) ret_payload;
```

Comment:
Ok, after a small chat with Hanno, I'm not removing it after all, since in his packing work this variable is used more that once, which makes it more useful. Also, we think there is value in distinguishing between potentially-negative return values and actual sizes.

(If we could use C99-style declarations I would write the code as follows, but unfortunately we can't yet. However once compiled it should be the same.)
```c
            const int ret_payload = mbedtls_ssl_get_max_out_record_payload( ssl );

            if( ret_payload < 0 )
            {
                MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_get_max_out_record_payload",
                                          ret_payload );
                return( ret_payload );
            }

            const size_t max_record_payload = (size_t) ret_payload;
```